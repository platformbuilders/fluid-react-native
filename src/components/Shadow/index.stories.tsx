import React from 'react';
import { Text } from 'react-native';
import type { Meta } from '@storybook/react';
import Shadow from '.';

const meta: Meta<typeof Shadow> = {
  title: 'Shadow',
  component: Shadow,
  args: {
    hasShadow: true,
    children: <Text>Shadow</Text>,
  },
  parameters: {
    notes: 'Este componente Ã© utilizado para renderizar um shadow.',
  },
};

export default meta;

export const Default = {};
