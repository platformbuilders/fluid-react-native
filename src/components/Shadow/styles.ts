import styled from 'styled-components/native';

export const View = styled.View`
  shadow-offset: 0px 4px;
  shadow-opacity: 0.2;
  elevation: 3;
  shadow-radius: 4px;
`;
