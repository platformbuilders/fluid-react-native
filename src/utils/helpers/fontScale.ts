import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
