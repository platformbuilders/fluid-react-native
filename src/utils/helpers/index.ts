export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
