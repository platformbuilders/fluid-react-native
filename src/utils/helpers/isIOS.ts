import { Platform } from 'react-native';

const IOS_PLATFORM = 'ios';

export const isIOS = (): boolean => Platform.OS === IOS_PLATFORM;
