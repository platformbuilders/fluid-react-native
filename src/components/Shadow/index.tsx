import React, { FC } from 'react';
import { View as DefaultView } from 'react-native';
import If from '../If';
import { View } from './styles';

type Props = {
  hasShadow?: boolean;
};

const Shadow: FC<Props> = ({ children, hasShadow = true, ...rest }) => (
  <>
    <If condition={hasShadow}>
      <View {...rest}>{children}</View>
    </If>
    <If condition={!hasShadow}>
      <DefaultView {...rest}>{children}</DefaultView>
    </If>
  </>
);

export default Shadow;
