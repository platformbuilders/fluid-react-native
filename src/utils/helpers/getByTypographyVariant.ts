import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): number => {
  const fontSize = getTheme(`${props.variant}.fontSize`)(props) || 16;
  return scaledFontSize(fontSize);
};

export const getLineHeight = (props: any): number => {
  const lineHeight = getTheme(`${props.variant}.lineHeight`)(props) || 22;
  return scaledFontSize(lineHeight);
};
