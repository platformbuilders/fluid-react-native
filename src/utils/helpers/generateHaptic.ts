import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
