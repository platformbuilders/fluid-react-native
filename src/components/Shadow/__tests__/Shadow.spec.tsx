import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import Shadow from '..';
import { theme } from '../../../test/helpers';

describe('<Shadow />', () => {
  it('should render Shadow correctly', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Shadow />
      </ThemeProvider>,
    );

    expect(render.toJSON()).toMatchSnapshot();
  });
});
