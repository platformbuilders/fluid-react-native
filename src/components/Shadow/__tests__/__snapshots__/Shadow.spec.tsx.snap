// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shadow /> should render Shadow correctly 1`] = `
<View
  style={
    Object {
      "elevation": 2,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.28,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
/>
`;
